chain/network/src/types.rs

```diff
@@ -742,6 +759,65 @@ pub struct NetworkConfig {
     pub outbound_disabled: bool,
 }
 
+impl NetworkConfig {
+    pub fn verify(&self) -> Result<(), String> {
+        let mut s = String::new();
+
+        if self.ideal_connections_lo > self.ideal_connections_hi {
+            writeln!(
+                s,
+                "Invalid IDEAL_CONNECTIONS values. LO({}) > HI({})",
+                self.ideal_connections_lo, self.ideal_connections_hi
+            )
+            .unwrap();
```

Comment:
The logic here feels strange. Shouldn't you return immediately when encountering an error?

Reply:
This was not intended to be a hard error, but just a description of all the errors found on the configuration. So I prefer to aggregate all the errors. Another option is just log them and don't return anything from this function.

Reply:
It seems to me that this should be a hard error. Otherwise there might be some undesirable consequences.